### PR TITLE
feat(auth_except): raise error if user is not logged in

### PR DESCRIPTION
**Pull request by paulineribeyre**

### CODE REVIEW

File: pidgin/app.py [hunk 1]
```diff
@@ -113,9 +112,13 @@ def send_query(query_txt):
     query = {'query': query_txt}
     data = requests.post(api_url, headers={'Authorization': auth}, json=query).json()
 
-    return flask.jsonify(data)
+    no_auth_msg = 'No authentication is provided'
```
Comment (rudyardrichter) [inline pidgin/app.py]: I think there should be a more robust way to check for an error here than with this string matching. If the status code from the request is 403 that's probably sufficient to raise an auth error, or if the response looked something like `{"error": {"code": 403, "message": "..."}}` that would be ideal

File: pidgin/errors.py [hunk 2]
```diff
@@ -4,6 +4,12 @@ def __init__(self, message):
         self.code = 500
 
 
+class AuthenticationException(PidginException):
+    def __init__(self, message):
+        self.message = str(message)
+        self.code = 511
```
Comment (rudyardrichter) [inline pidgin/errors.py]: I'm actually not sure what a 511 is for, but I would expect the code for this would be either 401 or 403

**Review by rudyardrichter — CHANGES_REQUESTED** [review]

File: pidgin/app.py [hunk 3]
```diff
@@ -111,11 +110,15 @@ def send_query(query_txt):
...
     auth = flask.request.headers.get('Authorization')
     query = {'query': query_txt}
-    data = requests.post(api_url, headers={'Authorization': auth}, json=query).json()
+    response = requests.post(api_url, headers={'Authorization': auth}, json=query)
+    data = response.json()
+
+    if response.status_code == 403:
+        raise AuthenticationException(data['message'])
+
+    return data
```
Comment (rudyardrichter) [inline pidgin/app.py:119]: I think this still needs `flask.jsonify(data)`

**Review by rudyardrichter — CHANGES_REQUESTED** [review]

File: pidgin/app.py (hunk 3 above)
Comment (paulineribeyre) [reply pidgin/app.py:119]: Removing `flask.jsonify()` allows me to remove `.get_json()` later, i thought it would be more readable

**Review by rudyardrichter — APPROVED** [review]

File: pidgin/app.py (hunk 3 above)
Comment (rudyardrichter) [reply pidgin/app.py:119]: Yeah, that comment doesn't make sense, I misread

**Pull request merged**